lib/internal/preferences/FilePreferenceStore.ts

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright © 2019 Atomist, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { HandlerContext } from "@atomist/automation-client";
+import { PreferenceStoreFactory } from "@atomist/sdm";
+import * as fs from "fs-extra";
+import * as os from "os";
+import * as path from "path";
+import {
+    lock,
+    unlock,
+} from "proper-lockfile";
+import {
+    AbstractPreferenceStore,
+    Preference,
+} from "./AbstractPreferenceStore";
+
+interface PreferenceFile {
+    [key: string]: { value: string, ttl?: number };
```

Comment:
You could `type PreferenceFile = Record<string, { value: string, ttl?: number }>;`